Add configurable token and payload field names for middleware and injector (#102)
bakdata/fastapi-jwks: CONTRIBUTING.md

# Contributing

Thank you for considering contributing to fastapi-jwks! We appreciate your help in making this project better.

## How to contribute

1. Fork the repository
2. Create a new branch for your feature or bugfix
3. Make your changes
4. Write or adapt tests as needed
5. Update the documentation if necessary
6. Submit a pull request

## Development setup

1. Clone your fork:
   ```sh
   git clone https://github.com/YOUR_USERNAME/fastapi-jwks.git
   cd fastapi-jwks
   ```

2. Install dependencies using Poetry:
   ```sh
   poetry install
   ```

3. Install pre-commit hooks:
   ```sh
   pre-commit install
   ```

## Running tests

To run tests:
```sh
pytest
```

To run tests with coverage:
```sh
pytest --cov=fastapi_jwks
```

## Code style

We use the following tools to maintain code quality:
- `ruff` for both linting and formatting
- `pyright` for type checking
- `pre-commit` hooks to automate checks

Please make sure your code passes all checks before submitting a PR:
```sh
pre-commit run --all-files
```

## Pull Request Process

> 1. Create an issue describing the changes you want to make
> 2. Fork the repository and create a branch from `main`
> 3. Make your changes and commit them with clear, descriptive commit messages
> 4. Update the README.md if needed
> 5. Add or modify tests as needed
> 6. Push to your fork and submit a pull request
> 7. Wait for the review and address any comments

Reviewer: ```suggestion
We are happy if you want to contribute to this project. If you find any bugs or have suggestions for improvements, please open an issue. We are also happy to accept your PRs. Just open an issue beforehand and let us know what you want to do and why.
```

## License

> By contributing to fastapi-jwks, you agree that your contributions will be licensed under its MIT License.

Reviewer: Please use the text from Kpops, its the same for all of our repositories, e.g.: https://github.com/bakdata/streams-bootstrap?tab=readme-ov-file#contributing

> By contributing to fastapi-jwks, you agree that your contributions will be licensed under its MIT License.

Reviewer: ```suggestion
fastapi-jwks is licensed under the [MIT License](https://github.com/bakdata/fastapi-jwks/blob/main/LICENSE).
```

Reviewer: Okay, I didn't catch that Kpops has a separate contributing file. I was just talking about the top-level readme and I'd like to align their content

Author: I accept them. Contributing part inside the root readme I think it was already the same as the others
